vernac/library.ml

```diff
@@ -57,14 +57,18 @@ let in_delayed f ch ~segment =
 
 let fetch_delayed del =
   let { del_digest = digest; del_file = f; del_off = pos; } = del in
-  try
-    let ch = open_in_bin f in
-    let () = LargeFile.seek_in ch pos in
-    let obj = System.marshal_in f ch in
-    let digest' = Digest.input ch in
-    if not (String.equal digest digest') then raise (Faulty f);
-    obj
-  with e when CErrors.noncritical e -> raise (Faulty f)
+  let ch = open_in_bin f in
+  let obj, digest' =
+    try
+      let () = LargeFile.seek_in ch pos in
+      let obj = System.marshal_in f ch in
```

Comment:
This can also fail if the file is corrupted, and you'll get an anomaly instead. I don't think it's a good idea not to catch this exception.

Reply:
That's just a user_err

Reply:
What about the other functions? `seek_in` and `Digest.input` can probably fail as well.

Reply:
Let them fail, we don't gain anything by catching them

Reply:
You'll get an anomaly instead of an error, in which case I don't really see the point of the `Faulty` exception and its accompanying message.